torch/csrc/jit/codegen/cuda/scheduler/pointwise.cpp

```diff
@@ -841,84 +856,33 @@ void schedulePointwise(Fusion* fusion, const PointwiseParams& params) {
-      // input into a consumer
-      input_cache_and_consumer.emplace_back(
-          std::make_pair(cached_input, consumer_tvs[0]));
-
-      // Grab all consumers which will be used for inlining computeAt for the
-      // body of the computation (excluding caching inputs/outputs)
-      for (auto consumer_tv : consumer_tvs) {
-        // Don't duplicate
-        if (added.insert(consumer_tv).second) {
-          consumers_of_cached_inputs.emplace_back(consumer_tv);
-        }
-      }
-    }
-  }
-
-  for (auto entry : input_cache_and_consumer) {
-    // Compute at inside unswitch position:
-    auto input_cache = entry.first;
-    auto input_cache_consumer = entry.second;
-
-    auto unswitch_it = std::find_if(
-        input_cache_consumer->domain()->domain().begin(),
-        input_cache_consumer->domain()->domain().end(),
-        [](IterDomain* id) {
-          return id->getParallelType() == ParallelType::Unswitch;
-        });
-    auto unswitch_pos =
-        unswitch_it == input_cache_consumer->domain()->domain().end()
-        ? -1
-        : std::distance(
-              input_cache_consumer->domain()->domain().begin(), unswitch_it) +
-            1;
+  // Begin by inlining at the unswitch position for the entire DAG. The cached
+  // inputs, outputs and cached outputs will keep this inline position, but
+  // other tensors will get a higher position in later inline propagation.
+  InlinePropagator inline_unswitch(
+      reference_tv, unswitch_pos, ComputeAtMode::BestEffort);
+  spanning_tree.traverse(&inline_unswitch);
 
-    input_cache->computeAt(
-        input_cache_consumer, unswitch_pos, ComputeAtMode::BestEffort);
+  // Inline at the inner most position. The CA position of all tensors except
+  // inputs, cached inputs, outputs and cached outputs will be updated.
+  std::unordered_set<TensorView*> inner_most_tensors(
+      all_tvs.begin(), all_tvs.end());
+  for (auto cached_input : cached_inputs) {
+    inner_most_tensors.erase(cached_input);
   }
-
-  // Producers for inlined computeAt
-  std::vector<TensorView*> compute_from = consumers_of_cached_inputs;
-
-  // Consumers for inlined computeAt
-  std::vector<TensorView*> compute_to;
-  // Compute at cached outputs
-  //[BIDx, Unswitch, Vectorization, TIDx]
   for (auto entry : cached_outputs) {
     auto cached_output = entry.first;
     auto output = entry.second;
-
-    auto unswitch_it = std::find_if(
-        output->domain()->domain().begin(),
-        output->domain()->domain().end(),
-        [](IterDomain* id) {
-          return id->getParallelType() == ParallelType::Unswitch;
-        });
-    auto unswitch_pos = unswitch_it == output->domain()->domain().end()
-        ? -1
-        : std::distance(output->domain()->domain().begin(), unswitch_it) + 1;
-
-    cached_output->computeAt(output, unswitch_pos, ComputeAtMode::BestEffort);
-    compute_to.push_back(cached_output);
+    inner_most_tensors.erase(cached_output);
```

Comment:
Can't you leave cached_output in for this next traversal? I think it's just the output we need to protect as it will have the vectorized dimension.

Reply:
I was thinking about the same, but if I did so there would be two failures:
```
[  FAILED  ] 2 tests, listed below:
[  FAILED  ] NVFuserTest.FusionParser_CUDA
[  FAILED  ] NVFuserTest.FusionChannelsLastParser_CUDA
```
so I reverted back. So seems that the previous behavior was excluding `cached_output` from getting a new inline position as well.

But these two failures are trivial to fix and should have no real changes in the generated kernels. I pushed a new commit https://github.com/csarofeen/pytorch/pull/1815/commits/24397fbfb3712a38d8af1f9266f28fd3052bdb39 for this, please check if this makes sense to you.